projects/storefrontapp/src/app/app.module.ts

```diff
@@ -62,6 +67,7 @@ if (!environment.production) {
   ],
   providers: [
     provideHttpClient(withFetch(), withInterceptorsFromDi()),
+    provideClientHydration(withEventReplay(), withNoHttpTransferCache()),
```

Comment:
If we decide to put `withNoHttpTransferCache()` in our storefrontapp, we should also:
1. [before mergin this PR] add it to customer's fresh apps (via the logic of our installation schematics)
2. [after merging this PR] document how to enable non-destructive hydration, mentioning that `withNoHttpTransferCache` is necessary and why
   - I guess it would be good to mention it somewhere in our official docs of SSR https://github.com/SAP/spartacus-docs)
   - and maybe also in a possible announcement published by Matthew in SAP notes at the day of the release (to be discussed with Matt)